internal/pkg/aws/codepipeline/codepipeline.go

```
@@ -210,6 +212,39 @@ func (c *CodePipeline) ListPipelineNamesByTags(tags map[string]string) ([]string
 	return pipelineNames, nil
 }
 
+// ListPipelineExecution returns the ExecutionID of the most recent execution of a pipeline.
+func (c *CodePipeline) ListPipelineExecution(pipelineName string) (*string, error) {
+	input := &cp.ListPipelineExecutionsInput{
+		MaxResults:   aws.Int64(1),
+		PipelineName: &pipelineName,
+	}
+	output, err := c.client.ListPipelineExecutions(input)
+	if err != nil {
+		return nil, fmt.Errorf("list pipeline execution for %s: %w", pipelineName, err)
+	}
+	return output.PipelineExecutionSummaries[0].PipelineExecutionId, nil
```

Will this ever return a zero length array?

---

Yeah and before return maybe worth doing a double check that `pipelineName` and output pipeline name match, in case `pipelineName` is "foo" and it might return pipeline with the name "foobar" when "foo" doesn't exist. Also instead of returning `*string`, I think `string` would be much easier to use and return error if the execution ID is not found.

---

I added a zero-length check. 
Discussed name-check w/ @iamhopaul123 offline and because the pipeline name isn't part of ListPipelineExecutionsOutput, it's not worth another API call to GetPipelineExecution to confirm.